operator/src/main/java/oracle/kubernetes/weblogic/domain/model/DomainCreationImage.java

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.weblogic.domain.model;
+
+import java.util.Optional;
+
+import jakarta.validation.constraints.NotNull;
+import oracle.kubernetes.common.utils.CommonUtils;
+import oracle.kubernetes.json.Description;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+public class DomainCreationImage implements DeploymentImage {
```

Comment:
I thought about this, but the field names, descriptions, etc. in DomainCreationImape need to be different than AuxiliaryImage since we don't want users to confuse the new feature with the existing MII-only auxiliary images feature. Only the model class is different, but other code is shared.